compiler/optimizer/Structure.cpp

```diff
@@ -184,7 +186,788 @@ void TR_RegionStructure::cleanupAfterEdgeRemoval(TR::CFGNode *node)
       }
    }
 
+// Implementation of TR_RegionStructure::extractUnconditionalExits() below
+class TR_RegionStructure::ExitExtraction
+   {
+   public:
+   ExitExtraction(TR::Compilation * const comp, TR::Region &memRegion)
+      : _comp(comp)
+      , _cfg(_comp->getFlowGraph())
+      , _trace(_comp->getOption(TR_TraceExitExtraction))
+      , _memRegion(memRegion)
+      , _structureRegion(_cfg->structureRegion())
+      , _workStack(_memRegion)
+      , _queued(std::less<TR_Structure*>(), _memRegion)
+      , _regionContents(RcCmp(), _memRegion)
+      , _allPreds(_memRegion)
+      , _succs(_memRegion)
+      , _excSuccs(_memRegion)
+      { }
+
+   void extractUnconditionalExits(const TR::list<TR::Block*, TR::Region&> &blocks);
+
+   private:
+   typedef TR::typed_allocator<TR_Structure*, TR::Region&> StructureAlloc;
+   typedef std::vector<TR_Structure*, StructureAlloc> StructureVec;
+   typedef std::set<TR_Structure*, std::less<TR_Structure*>, StructureAlloc> StructureSet;
+
+   typedef TR::typed_allocator<TR::CFGEdge*, TR::Region&> EdgeVecAlloc;
+   typedef std::vector<TR::CFGEdge*, EdgeVecAlloc> EdgeVec;
+
+   typedef std::pair<TR_RegionStructure*, TR_BitVector> RcEntry;
```

Comment:
Any particular reason the TR_BitVectors are by value rather than just being allocated from the region and only storing the pointer to reduce copying?

Reply:
It just saves an indirection. [No bit vector contents are copied](https://github.com/jdmpapin/omr/blob/80480340ac05b8cd8da535e4843f452613e4aace/compiler/optimizer/Structure.cpp#L641-L644)

Reply:
Ok sounds good to me - thanks!